server/src/main/resources/org/elasticsearch/bootstrap/test-framework.policy

```diff
@@ -125,4 +125,6 @@ grant codeBase "${codebase.netty-transport}" {
 grant {
   permission java.net.SocketPermission "127.0.0.1", "connect, resolve";
   permission java.nio.file.LinkPermission "symbolic";
+  // needed for keystore tests
+  permission java.lang.RuntimePermission "accessUserInformation";
```

Comment:
Why wasn't this permission needed for the keystore tests before?

Reply:
We now bootstrap the keystore ServerCli, if needed, whereas before we used to create it for the first time in initPhase2 in Elasticsearch.java. So this causes the ServerCliTests to attempt to create the keystore and we need the permission above on the save of the keystore. Exception trace below:

```
java.security.AccessControlException: access denied ("java.lang.RuntimePermission" "accessUserInformation")

	at __randomizedtesting.SeedInfo.seed([EBEF74F3275F24B8:1EC4F6A5E56A7696]:0)
	at java.base/java.security.AccessControlContext.checkPermission(AccessControlContext.java:485)
	at java.base/java.security.AccessController.checkPermission(AccessController.java:1068)
	at java.base/java.lang.SecurityManager.checkPermission(SecurityManager.java:416)
	at java.base/sun.nio.fs.UnixFileAttributeViews$Posix.checkWriteExtended(UnixFileAttributeViews.java:195)
	at java.base/sun.nio.fs.UnixFileAttributeViews$Posix.setMode(UnixFileAttributeViews.java:264)
	at java.base/sun.nio.fs.UnixFileAttributeViews$Posix.setPermissions(UnixFileAttributeViews.java:299)
	at org.elasticsearch.common.settings.KeyStoreWrapper.save(KeyStoreWrapper.java:507)
	at org.elasticsearch.common.settings.KeyStoreWrapper.save(KeyStoreWrapper.java:434)
	at org.elasticsearch.common.settings.KeyStoreWrapper.bootstrap(KeyStoreWrapper.java:239)
	at org.elasticsearch.common.settings.KeyStoreLoader.bootstrap(KeyStoreLoader.java:34)
	at org.elasticsearch.server.cli.ServerCli.execute(ServerCli.java:89)
	at org.elasticsearch.common.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:56)
```

Reply:
Can we narrow the permission scope to the server codebase? There's an existing grant in this file already it can be added to:

```
grant codeBase "${codebase.elasticsearch}" {
```

Reply:
I don't think it's possible. I would have to add an AccessControl block and we don't install the security manager in ServerCli. The test framework is artificially enforcing the security manager in this code.

I think the right fix would be to add `@ESTestCase.WithoutSecurityManager` to the ServerCliTests, we don't actually have a security manager enabled in the ServerCli code. What do you think?

Reply:
That sounds right to me, no need for running the cli tests under security manager (in fact, you could just add that annotation to CommandTestCase).